rpcs3/Emu/RSX/Common/TextureUtils.cpp

```diff
@@ -24,12 +24,128 @@ namespace utils
 namespace
 {
 
+#ifndef __APPLE__
 u16 convert_rgb655_to_rgb565(const u16 bits)
 {
 	// g6 = g5
 	// r5 = (((bits & 0xFC00) >> 1) & 0xFC00) << 1 is equivalent to truncating the least significant bit
 	return (bits & 0xF81F) | (bits & 0x3E0) << 1;
 }
+#else
+u32 convert_rgb565_to_bgra8(const u16 bits)
+{
+	const u8 r5 = ((bits >> 11) & 0x1F);
```

Comment:
Uh, I am convinced that this needs at least be vectorized otherwise it'll be sluggish.

Reply:
Don't bother, it should be really trivial for auto-vectorization in clang/gcc (or at least with minimal changes)